examples/train_simple_navigation.py

```diff
@@ -0,0 +1,319 @@
+    prioritized_replay_alpha = 0.6
+    prioritized_replay_beta0 = 0.4
+    prioritized_replay_eps = 1e-3
+    target_net_update_freq = 10000
+    log_freq = 100
+    save_freq = 1000
+    action_discretize_levels = 6
+    resized_image_size = (84, 84)
+    model_path = '/tmp/learn_simple_navigation/agent.model'
+
+
+def main():
+    options = Options()
+    env = gym.make("SocialBot-SimpleNavigationNoLanguage-v0")
+    assert isinstance(env.action_space, gym.spaces.Box)
+    assert isinstance(env.observation_space, gym.spaces.Box)
+    image_shape = env.observation_space.shape
+    agent = QAgent(
+        image_shape=(image_shape[2], ) + options.resized_image_size,
+        num_actions=options.action_discretize_levels**2,
+        options=options)
+    rewards = deque(maxlen=options.log_freq)
+    steps = deque(maxlen=options.log_freq)
+    total_steps = 0
+    episodes = 0
+    while total_steps < options.max_steps:
+        reward, step = run_one_episode(env, agent, options)
+        rewards.append(reward)
+        steps.append(step)
+        total_steps += step
+        episodes += 1
+        if episodes % options.log_freq == 0:
+            logging.info("episodes=%s " % episodes +
+                         "total_steps=%s " % total_steps +
+                         "mean_reward=%s " % (sum(rewards) / len(rewards)) +
+                         "mean_steps=%s " % (sum(steps) / len(steps)) +
+                         "exp_rate=%s " % agent.get_exploration_rate())
+
+        if episodes % options.save_freq == 0:
+            agent.save_model(options.model_path)
+
+
+def run_one_episode(env, agent, options):
+    obs = env.reset()
+    agent.start_new_episode()
+    episode_reward = 0.
+    reward = 0
+    done = False
+    steps = 0
+
+    while not done:
+        obs = PIL.Image.fromarray(obs).resize(options.resized_image_size,
+                                              PIL.Image.ANTIALIAS)
+        obs = np.transpose(obs, [2, 0, 1])
+        action, q = agent.act(obs, reward)
+        control = [
+            0.02 * (action // options.action_discretize_levels),
+            0.02 * (action % options.action_discretize_levels), 0
+        ]
+        new_obs, reward, done, _ = env.step(control)
+        agent.learn(obs, action, reward, done)
+        obs = new_obs
+        episode_reward += reward
+        steps += 1
+    if options.log_freq == 1:
+        logging.info("reward=%s" % reward + " steps=%s" % steps + " q=%s" % q)
+    return episode_reward, steps
+
+
+Experience = namedtuple(
+    "Experience", field_names=["state", "action", "reward", "done"])
+
+
+class QAgent(object):
+    def __init__(self, image_shape, num_actions, options):
+        num_image_channels = image_shape[0]
+        num_input_channels = num_image_channels * (options.history_length + 1)
+        if options.f_action_feature is not None:
+            num_input_channels += len(
+                (options.f_action_feature)(0)) * options.history_length
```

Comment:
this does not seem to match signature of lambda function:
f_action_feature = lambda _, action

Reply:
This is because f_action_feature is treated as a class method and '_' is for self.